Packages/vcs/Lib/vcs2vtk.py

```diff
@@ -73,34 +73,61 @@ def numpy_to_vtk_wrapper(numpyArray, deep=False, array_type=None):
     return result
 
 
+# Adds 'array' to 'grid' as cell or point attribute based on 'isCellData'
+# It also sets it as the active scalar if 'isScalars'.
+# If the grid has pedigree ids (it was wrapped) we use them to set the array.
+def setArray(grid, array, arrayName, isCellData, isScalars):
+    attributes = grid.GetCellData() if isCellData else grid.GetPointData()
+    pedigreeId = attributes.GetPedigreeIds()
+    if (pedigreeId):
+        vtkarray = attributes.GetArray(arrayName)
+        for i in range(0, vtkarray.GetNumberOfTuples()):
+            vtkarray.SetValue(i, array[pedigreeId.GetValue(i)])
+    else:
+        vtkarray = numpy_to_vtk_wrapper(array, deep=False)
+        vtkarray.SetName(arrayName)
+        attributes.AddArray(vtkarray)
+    if (isScalars):
+        attributes.SetActiveScalars(arrayName)
+
+
 def putMaskOnVTKGrid(data, grid, actorColor=None, cellData=True, deep=True):
     # Ok now looking
     msk = data.mask
-    imsk = numpy_to_vtk_wrapper(msk.astype(numpy.int).flat, deep=deep)
     mapper = None
     if msk is not numpy.ma.nomask and not numpy.allclose(msk, False):
         if actorColor is not None:
+            flatIMask = msk.astype(numpy.int).flat
             if grid.IsA("vtkStructuredGrid"):
                 grid2 = vtk.vtkStructuredGrid()
+                vtkmask = numpy_to_vtk_wrapper(flatIMask, deep=deep)
+                attributes2 = grid2.GetCellData() if cellData else grid2.GetPointData()
             else:
                 grid2 = vtk.vtkUnstructuredGrid()
+                if (cellData):
+                    attributes2 = grid2.GetCellData()
+                    attributes = grid.GetCellData()
+                else:
+                    attributes2 = grid2.GetPointData()
+                    attributes = grid.GetPointData()
+                attributes2.SetPedigreeIds(attributes.GetPedigreeIds())
```

Comment:
It is a terminology VTK uses. It is just an index to keep track of where pieces of cells end up after I wrap the dataset around. Certain cells will be cut in two pieces which will end up in different parts of the dataset. When the scalars are updated I use those indexes to set the proper scalar for each cell.